manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -25,6 +25,7 @@ spec:
       labels:
         {{- include "gateway.sidecarInjectionLabels" . | nindent 8 }}
         {{- include "gateway.selectorLabels" . | nindent 8 }}
+        # FIXME
```

Comment:
wrt `app.kubernetes.io/name` I see a few options:
1. In `istio.labels` we have a line `app.kubernetes.io/name: {{ template "name" }}` and in each chart we define the `name` template in `_helpers.yaml`. I think this is the approach helm suggests. This reduces a lot of boilerplate but feels indirect.
2. Before every use of `{{ include "istio.labels" }}`, we have `app.kubernetes.io/name: chartname` (this is what I have done in this chart). This is the most direct, but feels repetitive
3. In every chart, we define a template that looks like
  ```yaml
{{ define mychart.labels }}
app.kubernetes.name: chartname
{{ include "istio.labels" . }}
{{ end }}
```
  This also feels a bit repetitive.

Reply:
> wrt `app.kubernetes.io/name` I see a few options:
> 1. In `istio.labels` we have a line `app.kubernetes.io/name: {{ template "name" }}` and in each chart we define the `name` template in `_helpers.yaml`. I think this is the approach helm suggests. This reduces a lot of boilerplate but feels indirect.
> 2. Before every use of `{{ include "istio.labels" }}`, we have `app.kubernetes.io/name: chartname` (this is what I have done in this chart). This is the most direct, but feels repetitive
> 3. In every chart, we define a template that looks like
>   ```yaml
> {{ define mychart.labels }}
> app.kubernetes.name: chartname
> {{ include "istio.labels" . }}
> {{ end }}
> ```
>   This also feels a bit repetitive.

I am fine with repeating some, especially if they actually DO vary by chart. It's the ones that never vary and can be stamped out that I want a consistent pattern for.

 2 seems pretty simple, and we can always improve that later - I am fine with it.